packages/chakra-ui/src/Stack/index.js

```diff
@@ -7,32 +7,26 @@ import Box from "../Box";
 const Stack = ({
   direction,
   isInline = false,
+  isReversed = false,
   children,
   align,
   justify,
   spacing = 2,
   shouldWrapChildren,
   ...rest
 }) => {
-  let flexDirection;
-
-  if (isInline != null) {
-    flexDirection = isInline ? "row" : "column";
-  }
-
-  if (direction != null) {
-    flexDirection = direction;
-  }
+  direction = direction || (isInline ? "row" : "column");
+  isInline |= direction.startsWith("row");
+  isReversed |= direction.endsWith("reverse");
```

Comment:
The bitwise operator looks more concise I'd like to keep it a bit verbose so a developer doesn't have to google what `|=` means in JavaScript. You get what I mean.

This works though.